modules/vpc/variables.tf

```diff
@@ -126,15 +132,20 @@ locals {
   #
   # _Note_: We need **actual name** to match real role, which means
   # `local.region` and not `local.iam_region`.
-  sls_lambda_role_name = "${local.sls_service_name}-${local.stage}-${local.region}-lambdaRole"
+  sls_lambda_role_default_name = "${local.sls_service_name}-${local.stage}-${local.region}-lambdaRole"
+
+  sls_lambda_role_name = "${local.lambda_role_name != "" ? local.lambda_role_name : local.sls_lambda_role_default_name}"
 
   # The built-in serverless Lambda execution role ARN.
   #
   # Note that we use `iam_region` to potentially wildcard the IAM permission
   # in the actual name of the role.
   #
   # - No region allowed in ARN. See https://iam.cloudonaut.io/reference/iam.html
-  sls_lambda_role_arn = "arn:${local.iam_partition}:iam::${local.iam_account_id}:role/${local.sls_service_name}-${local.stage}-${local.iam_region}-lambdaRole"
+  sls_lambda_role_default_arn = "arn:${local.iam_partition}:iam::${local.iam_account_id}:role/${local.sls_service_name}-${local.stage}-${local.iam_region}-lambdaRole"
+
+  sls_lambda_role_custom_arn = "arn:${local.iam_partition}:iam::${local.iam_account_id}:role/${local.lambda_role_name}"
+  sls_lambda_role_arn        = "${local.lambda_role_name != "" ? local.sls_lambda_role_custom_arn : local.sls_lambda_role_default_arn}"
```

Comment:
Can't we just do:

```hcl
sls_lambda_role_arn = "arn:${local.iam_partition}:iam::${local.iam_account_id}:role/${local.lambda_role_name}"
```

... as we've already done the same conditional check above...

Reply:
I thought so, but the default ARN includes `iam_region`, which AFAICT is used to optionally wildcard regions?

Reply:
Oh nuts, you're right. Carry on...